application/src/main/java/md/maib/retail/application/find_campaign_by_metainfo/FindCampaignByMetaInfoService.java

```diff
@@ -0,0 +1,28 @@
+package md.maib.retail.application.find_campaign_by_metainfo;
+
+import md.maib.retail.application.CampaignAllInfo;
+import md.maib.retail.model.ports.Campaigns;
+
+import java.util.Objects;
+import java.util.Optional;
+
+ public class FindCampaignByMetaInfoService implements FindCampaignByMetaInfoUseCase{
+    private final Campaigns campaigns;
+
+    public FindCampaignByMetaInfoService(Campaigns campaigns) {
+        this.campaigns = Objects.requireNonNull(campaigns,"Campaigns must not be null");
+    }
+
+    @Override
+    public Optional<CampaignAllInfo> findByMetaInfo(String key, String value) {
```

Comment:
findByMetaInfo can return many campaigns.
Example: all campaigns of Felicia: by brands Farmacia and Optica, standard and Promo

Reply:
i should make as a list of campaigns?